frontend/src/components/MapView/DateSelector/TimelineItems/index.tsx

```diff
@@ -114,29 +137,95 @@ const TimelineItems = memo(
       [classes.dateItemLabel, classes.dayItem, locale],
     );
 
+    const hasValidityDates = (
+      selectedLayersList: DateCompatibleLayer[],
+      layerIndex: number,
+      date: DateRangeType,
+    ) => {
+      return (
+        selectedLayersList &&
+        selectedLayersList[layerIndex] &&
+        selectedLayersList[layerIndex].validity &&
+        (selectedLayersList[layerIndex] as AdminLevelDataLayerProps).dates &&
+        (selectedLayersList[
+          layerIndex
+        ] as AdminLevelDataLayerProps).dates!.includes(date.date)
+      );
+    };
+
     const renderLayerDates = useCallback(
       (date: DateRangeType, index: number) => {
+        const isValidityBackwardOrBothDirection = (
```

Comment:
This feels like we're duplicating a lot of the logic from the validity dates.
Could we rethink a bit how we format dates and maybe do a check that looks like:

displayDate == valueDate and then just fetch the validity mode for that layer?

@HarryMytilinaios since you've worked on the dates quite a bit recently it would be nice to have your input as well :)

Reply:
I did not understand how you want it refactored. What logic is duplicated ?